include/openthread/ip6.h

```diff
@@ -409,9 +409,18 @@ void otIp6SetReceiveFilterEnabled(otInstance *aInstance, bool aEnabled);
 /**
  * This function sends an IPv6 datagram via the Thread interface.
  *
+ * The caller transfers ownership of @p aMessage when making this call. OpenThread will free @p aMessage when
+ * processing is complete, including when a value other than `OT_ERROR_NONE` is returned.
+ *
  * @param[in]  aInstance A pointer to an OpenThread instance.
  * @param[in]  aMessage  A pointer to the message buffer containing the IPv6 datagram.
  *
+ * @retval OT_ERROR_NONE      Successfully processed the message.
+ * @retval OT_ERROR_DROP      Message was dropped during processing.
```

Comment:
DROP seems to mean that the packet is well formed, but somehow inappropriate.  Is that right?  Is there some way to convey that here?

Reply:
You are correct. One example is a packet containing an unknown IPv6 Hop-by-Hop Option with an Option Type that specifies discarding the message when the option is unknown.

How about something like: "Message was well-formed but not fully processed due to packet processing rules."

Reply:
That sounds good.